src/Logger/Adapter/NewRelic.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Utopia\Logger\Adapter;
+
+use Exception;
+use Utopia\Logger\Adapter;
+use Utopia\Logger\Log;
+
+// Reference Material
+// https://docs.newrelic.com/docs/logs/log-api/introduction-log-api
+
+class NewRelic extends Adapter
+{
+    /**
+     * @var string (required, can be found in NewRelic -> Account dropdown -> API keys)
+     */
+    protected string $licenseKey;
+
+    /**
+     * @var string (optional, the api url where New Relic is reachable, in case of an EU New Relic account could
+     *              is 'https://log-api.eu.newrelic.com/log/v1'. defaults to 'https://log-api.newrelic.com/log/v1')
+     */
+    private string $apiUrl = 'https://log-api.newrelic.com/log/v1';
+
+    /**
+     * Return unique adapter name
+     *
+     * @return string
+     */
+    public static function getName(): string
+    {
+        return "newRelic";
+    }
+
+    /**
+     * Push log to external provider
+     *
+     * @param Log $log
+     * @return int
+     * @throws Exception
+     */
+    public function push(Log $log): int
+    {
+        $params = [];
+        foreach ($log->getExtra() as $paramKey => $paramValue) {
+            $params[$paramKey] = var_export($paramValue, true);
+        }
+
+        $breadcrumbsArray = [];
+        foreach ($log->getBreadcrumbs() as $breadcrumb) {
+            $breadcrumbsArray[] = [
+                'timestamp' => (int) $breadcrumb->getTimestamp(),
+                'category' => $breadcrumb->getCategory(),
+                'action' => $breadcrumb->getMessage(),
+                'metadata' => [
+                    'type' => $breadcrumb->getType(),
+                ],
+            ];
+        }
+
+        $tags = [];
+        foreach ($log->getTags() as $tagKey => $tagValue) {
+            $tags[$tagKey] = $tagValue;
+        }
+
+        $requestBody = [
+            'timestamp' => (int) $log->getTimestamp(),
+            'message' => $log->getMessage(),
+            'logtype' => $log->getType(),
+            'version' => $log->getVersion(),
+            'environment' => $log->getEnvironment(),
+            'action' => $log->getAction(),
+            'namespace' => $log->getNamespace(),
+            'server' => $log->getServer(),
+            'user' => [
+                'id' => $log->getUser()->getId(),
+                'email' => $log->getUser()->getEmail(),
+                'username' => $log->getUser()->getUsername(),
+            ],
+            'breadcrumbs' => $breadcrumbsArray,
+            'tags' => $tags,
+            'params' => $params,
```

Comment:
how are `params` handled in new relic? I couldn't find this in the docs.

Reply:
Should this be here if it's processed down below?

Reply:
I've updated the description, there you can see how `params` are handled.
What exactly do you mean with it being processed down below?

Reply:
> What exactly do you mean with it being processed down below?

Never mind 😅